### PR TITLE
test: scm: convert to dir helpers

### PR DESCRIPTION
**Pull request by efiop**
Related to #2896

* [x] ❗ Have you followed the guidelines in the [Contributing to DVC](https://dvc.org/doc/user-guide/contributing/core) list?

* [x] 📖 Check this box if this PR **does not** require [documentation](https://dvc.org/doc) updates, or if it does **and** you have created a separate PR in [dvc.org](https://github.com/iterative/dvc.org) with such updates (or at least opened an issue about it in that repo). Please link below to your PR (or issue) in the [dvc.org](https://github.com/iterative/dvc.org) repo.

* [x] ❌ Have you checked DeepSource, CodeClimate, and other sanity checks below? We consider their findings recommendatory and don't expect everything to be addressed. Please review them carefully and fix those that actually improve code or fix bugs.

Thank you for the contribution - we'll try to review it as soon as possible. 🙏

### CODE REVIEW

File: tests/func/test_scm.py [hunk 1]
```diff
@@ -61,112 +62,95 @@ def test_commit_in_submodule(self):
...
 
 
-class TestIgnore(object):
-    @staticmethod
-    def _count_gitignore_entries(string):
-        lines = get_gitignore_content()
-        return len([i for i in lines if i == string])
+def _count_gitignore_entries(string):
+    lines = get_gitignore_content()
+    return len([i for i in lines if i == string])
```
Comment (Suor) [inline tests/func/test_scm.py]:
```suggestion
    return lines.count(string)
```

Comment (Suor) [reply tests/func/test_scm.py]: And maybe `string` -> `line`, since we only match full lines.

File: tests/func/test_scm.py [hunk 2]
```diff
@@ -61,112 +62,95 @@ def test_commit_in_submodule(self):
...
-        data_dir = os.path.join(
-            repo_dir._root_dir, os.path.join("dir1", "file1")
-        )
-        entry, gitignore = Git(repo_dir._root_dir)._get_gitignore(data_dir)
-        assert entry == "/file1"
-        assert gitignore == os.path.join(
-            repo_dir._root_dir, "dir1", Git.GITIGNORE
-        )
+    assert entry == "/dir"
+    assert gitignore == fspath(tmp_dir / ".gitignore")
```
Comment (Suor) [inline tests/func/test_scm.py]: Since there is no file nor dir this is simply duplication.

Comment (Suor) [reply tests/func/test_scm.py]: The whole test is checking some private thing, should we do that at all? Maybe ignore this for now.

File: tests/func/test_scm.py [hunk 3]
```diff
@@ -61,112 +62,95 @@ def test_commit_in_submodule(self):
...
-            assert l.endswith("\n")
+    scm.ignore(bar_path)
+    lines = gitignore.read_text().splitlines()
+    assert len(lines) == 2
+    assert gitignore.read_text().endswith("\n")
 
-        assert lines[0].strip() == foo_ignore_pattern
-        assert lines[1].strip() == bar_ignore_pattern
+    assert lines[0].strip() == foo_ignore_pattern
+    assert lines[1].strip() == bar_ignore_pattern
```
Comment (Suor) [inline tests/func/test_scm.py]: Won't that work?
```python
assert gitignore.read_text() == "/foo\n/bar\n"
```
Or at least:
```python
assert lines == ["/foo", "/bar"]
```
`.splitlines()` should already do the strip and no need to check for len this way. BTW, I prefer hard-coding strings like `/foo` and `/bar` here as you see, makes a test more obvious until it short enough)

File: tests/func/test_scm.py (hunk 2 above)
Comment (efiop) [reply tests/func/test_scm.py]: Yep, we should, but in unit tests. Out of focus for this PR though, as I've stated above.

File: tests/func/test_scm.py (hunk 3 above)
Comment (efiop) [reply tests/func/test_scm.py]: > Won't that work?

Not sure,  might not work because of CRLF.

File: tests/func/test_scm.py [hunk 4]
```diff
@@ -61,112 +62,93 @@ def test_commit_in_submodule(self):
...
-        assert os.path.isfile(Git.GITIGNORE)
-        assert self._count_gitignore_entries(target) == 1
+    foo = fspath(tmp_dir / "foo")
+    target = "/foo"
 
-        git.ignore(foo)
-        assert os.path.isfile(Git.GITIGNORE)
-        assert self._count_gitignore_entries(target) == 1
+    scm.ignore(foo)
+    assert os.path.isfile(Git.GITIGNORE)
```
Comment (pared) [inline tests/func/test_scm.py]: in latter tests, we use ".gitignore", lets use it or change the rest into `Git.GITIGNORE`

File: tests/func/test_scm.py [hunk 5]
```diff
@@ -61,112 +62,93 @@ def test_commit_in_submodule(self):
...
+    assert _count_gitignore_entries(target) == 1
 
-        git.ignore_remove(foo)
-        assert self._count_gitignore_entries(target) == 0
+    scm.ignore(foo)
+    assert os.path.isfile(Git.GITIGNORE)
```
Comment (pared) [inline tests/func/test_scm.py]: As above

File: tests/func/test_scm.py (hunk 4 above)
Comment (pared) [reply tests/func/test_scm.py]: we can also replace `os.path.isifle` -> (tmp_dir / ".gitignore").is_file(), but it does not seem it will change much.

**Pull request merged**